lxd/storage/drivers/driver_alletra_wsapi1.go

```diff
@@ -483,3 +533,91 @@ func (p *alletraClient) deleteHost(hostName string) error {
 func (p *alletraClient) updateHost(hostName string, qns []string) error {
 	return fmt.Errorf("Failed to update host %q. Operation not supported.", hostName)
 }
+
+func (p *alletraClient) _createVolume(poolName string, volName string, sizeBytes int64) error {
+	req := map[string]any{
+		"name":    volName,
+		"cpg":     p.driver.config["alletra.wsapi.cpg"],
+		"sizeMiB": sizeBytes / 1024 / 1024,
+		"tpvv":    true, // thinly provisioned volume
+	}
+
+	url := api.NewURL().Path("api", "v1", "volumes")
+	err := p.requestAuthenticated(http.MethodPost, url.URL, req, nil)
+	if err != nil {
+		return fmt.Errorf("Failed to create volume %q in storage pool %q: %w", volName, p.driver.name, err)
+	}
+
+	return nil
+}
+
+// createVolume creates a new volume in the given storage pool. The volume is created with
+// supplied size in bytes. Upon successful creation, volume's ID is returned.
+func (p *alletraClient) createVolume(poolName string, volName string, sizeBytes int64) error {
+	err := p._createVolume(poolName, volName, sizeBytes)
+	if err != nil {
+		return err
+	}
+
+	// Add a newly created volume to a volume set
+	err = p.modifyVolumeSet(1, volName) // 1 = memAdd
+	return err
+}
+
+// getVolume returns the volume for a given volName.
+func (p *alletraClient) getVolume(poolName string, volName string) (*hpeVolume, error) {
+	var resp hpeVolume
+
+	url := api.NewURL().Path("api", "v1", "volumes", volName)
+	err := p.requestAuthenticated(http.MethodGet, url.URL, nil, &resp)
+	if err != nil {
+		hpeErr, ok := err.(*hpeError)
+		if ok {
+			switch hpeErr.Code {
+			case 23: // NON_EXISTENT_VOL
+				return nil, api.StatusErrorf(http.StatusNotFound, "Volume/snapshot %q not found", volName)
+			default:
+				return nil, fmt.Errorf("HPE debug. hpeErr.Code: %d. hpeErr.Desc: %s", hpeErr.Code, hpeErr.Desc)
+			}
+		}
+
+		return nil, fmt.Errorf("Failed to get hpeVolume %q: %w", volName, err)
+	}
+
+	if resp.Name == "" {
+		return nil, fmt.Errorf("HPE volume %q exists but returned an empty response", volName)
+	}
+
+	return &resp, nil
+}
+
+func (p *alletraClient) _deleteVolume(poolName string, volName string) error {
+	url := api.NewURL().Path("api", "v1", "volumes", volName)
+
+	err := p.requestAuthenticated(http.MethodDelete, url.URL, nil, nil)
+	if err != nil {
+		hpeErr, ok := err.(*hpeError)
```

Comment:
I'm seeing this pattern duplicated a fair bit, sounds like we could do with a helper function that detects specific errors, similar to `api.StatusErrorCheck`